packages/nx/src/plugins/js/package-json/create-package-json.ts

```diff
@@ -103,44 +116,64 @@ export function createPackageJson(
       // don't store dev dependencies for production
       if (!options.isProduction) {
         packageJson.devDependencies ??= {};
-        packageJson.devDependencies[packageName] = version;
+        packageJson.devDependencies[packageName] = getVersion(
+          packageName,
+          version,
+          'devDependencies'
+        );
       }
     } else {
       if (!packageJson.peerDependencies?.[packageName]) {
         packageJson.dependencies ??= {};
-        packageJson.dependencies[packageName] = version;
+        packageJson.dependencies[packageName] = getVersion(
+          packageName,
+          version,
+          'dependencies'
+        );
       }
     }
   });
-  Object.entries(npmDeps.peerDependencies).forEach(([packageName, version]) => {
-    if (!packageJson.peerDependencies?.[packageName]) {
-      if (rootPackageJson.dependencies?.[packageName]) {
-        packageJson.dependencies ??= {};
-        packageJson.dependencies[packageName] = version;
-        return;
-      }
+  if (!isLibrary) {
```

Comment:
I think it's fine for now, but if the app vs lib logic deviates more, we might consider separating them into two functions: `createAppPackageJson` and `createLibPackgeJson` so the consumer can choose which to call depending on use-case.